dist/bin/scala

```diff
@@ -28,10 +28,39 @@ fi
 
 source "$PROG_HOME/bin/common"
 
+case `uname` in
+CYG*|MINGW*|MSYS*) PSEP=';' ;;
+*) PSEP=':' ;;
+esac
+
+ARGS=()
+while [ $# -gt 0 ]; do
+  case "$1" in
+  -cp | -classpath) # partial fix for #10761
+    # passed as two arguments, e.g. '-classpath' 'lib/*'
+    ARGS+=($1)
+    ARGS+=("$2${PSEP}")
+    shift
+    shift
+    ;;
+  -cp*|-classpath*) # partial fix for #10761
+    # passed as a single argument, e.g. '-classpath lib/*'
+    # (hashbang line can glom args together)
+    ARGS+=('-classpath')
+    ARGS+=("\"${1#* *}${PSEP}\"")
+    shift
+    ;;
+  *)
+    ARGS+=($1)
+    shift
+    ;;
+  esac
+done
```

Comment:
Is it really necessary to keep it in bash script and not in scala code?

Reply:
Why do we remove the compiler libraries on line 180?

Reply:
We remove the compiler conditionally. We reuse classpath from coursier, (or script) which bring all the classes on classpath, then filter them out by `removeCompiler` when there is no `-with-compiler` flag. Actually I did a quick look at the issue with the tests and missed the point we gather the classpath. The correct solution should be splitting and flattening the classpath while parsing args at 108 line I guess, so all further processing should have well defined classpath property.